### PR TITLE
Fix aws_lightsail_instance name validation

### LINKED ISSUE #29899: [Bug]: `aws_lightsail_instance` incorrectly requires instances to start with an alphabetical character
### Terraform Core Version

1.3.6

### AWS Provider Version

4.57.1

### Affected Resource(s)

* aws_lightsail_instance

### Expected Behavior

Instance `123-test` should be created

### Actual Behavior

Instance is not created. Provider produces the output below.

### Relevant Error/Panic Output Snippet

```shell
╷
│ Error: invalid value for name (must begin with an alphabetic character)
│ 
│   with aws_lightsail_instance.instance,
│   on main.tf line 15, in resource "aws_lightsail_instance" "instance":
│   15:   name              = "123-test"
│
```

### Terraform Configuration Files

```
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 3.0"
    }
  }
}

provider "aws" {
  region = "ap-southeast-2"
}

resource "aws_lightsail_instance" "instance" {
  name              = "123-test"
  availability_zone = "ap-southeast-2a"
  blueprint_id      = "centos_7_2009_01"
  bundle_id         = "medium_2_2"
}
```

### Steps to Reproduce

1. `terraform validate`

### Debug Output

_No response_

### Panic Output

n/a

### Important Factoids

Creating an instance with the above name works in both the Lightsail web management interface and in `aws-cli`

### References

* [https://docs.aws.amazon.com/lightsail/2016-11-28/api-reference/API_CreateInstances.html] - Note there is no requirement mentioned for instances to start with an alphabetic character

### Would you like to implement a fix?

Yes

### PR DESCRIPTION
**Pull request by danielcotton**
### Description

This PR corrects the overzealous instance name validation in `aws_lightsail_instance`. Previously the first character of an instance name had to be alphabetic; but can now be alphanumeric. I've added a positive test case for an instance with a numeric first character, and a negative test case for an instance with a symbol first character.
Due to the removal of the `amazon_linux` Lightsail blueprint, I have also updated all `aws_lightsail_instance` acceptance tests to use the `amazon_linux_2` blueprint.

### Relations

Closes #29899

### References

https://docs.aws.amazon.com/lightsail/2016-11-28/api-reference/API_CreateInstances.html - Note there is no requirement mentioned for instances to start with an alphabetic character

### Output from Acceptance Testing

```
$ make testacc TESTS=TestAccLightsailInstance_name PKG=lightsail
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/lightsail/... -v -count 1 -parallel 20 -run='TestAccLightsailInstance_name'  -timeout 180m
=== RUN   TestAccLightsailInstance_name
=== PAUSE TestAccLightsailInstance_name
=== CONT  TestAccLightsailInstance_name
--- PASS: TestAccLightsailInstance_name (246.28s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/lightsail	249.652s
...
```

### CODE REVIEW

File: internal/service/lightsail/instance_test.go [hunk 1]
```diff
@@ -75,8 +76,18 @@ func TestAccLightsailInstance_name(t *testing.T) {
 				ExpectError: regexp.MustCompile(`must contain only alphanumeric characters, underscores, hyphens, and dots`),
 			},
 			{
-				Config:      testAccInstanceConfig_basic(rNameWithStartingDigit),
-				ExpectError: regexp.MustCompile(`must begin with an alphabetic character`),
+				Config:      testAccInstanceConfig_basic(rNameWithStartingHyphen),
+				ExpectError: regexp.MustCompile(`must begin with an alphanumeric character`),
+			},
+			{
+				Config: testAccInstanceConfig_basic(rNameWithStartingDigit),
```
Comment (bschaatsbergen) [inline internal/service/lightsail/instance_test.go:83]: I would expect this test to fail?

**Comment by bschaatsbergen** [comment]
Could you please add a changelog entry to your PR :)

**Comment by danielcotton** [comment]
> Could you please add a changelog entry to your PR :)

Done & rebased :) 

Thanks for the quick response!

**Comment by bschaatsbergen** [comment]
Could you run the tests too and copy the output to the PR description please:

Similar to this PR description: https://github.com/hashicorp/terraform-provider-aws/pull/29224#issue-1568797448

**Comment by danielcotton** [comment]
I believe I already have? That's the ones under 'Output from Acceptance Testing', right?

**Comment by bschaatsbergen** [comment]
> I believe I already have? That's the ones under 'Output from Acceptance Testing', right?

My bad, phone didn't properly show it :)

File: internal/service/lightsail/instance_test.go (hunk 1 above)
Comment (bschaatsbergen) [reply internal/service/lightsail/instance_test.go:83]: 👍🏼

**Review by bschaatsbergen — APPROVED** [review]
LGTM 🚀 

```
 make testacc TESTS=TestAccLightsailInstance_name PKG=lightsail                                                            
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/lightsail/... -v -count 1 -parallel 20 -run='TestAccLightsailInstance_name'  -timeout 180m
=== RUN   TestAccLightsailInstance_name
=== PAUSE TestAccLightsailInstance_name
=== CONT  TestAccLightsailInstance_name
--- PASS: TestAccLightsailInstance_name (188.56s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/lightsail  188.642s
```

**Comment by bschaatsbergen** [comment]
Thanks for your contribution 👌🏼 @danielcotton

**Comment by bschaatsbergen** [comment]
I'll ask @ewbankkit to sign this off.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccLightsailInstance_name\|TestAccLightsailInstance_basic' PKG=lightsail ACCTEST_PARALLELISM=3
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/lightsail/... -v -count 1 -parallel 3  -run=TestAccLightsailInstance_name\|TestAccLightsailInstance_basic -timeout 180m
=== RUN   TestAccLightsailInstance_basic
=== PAUSE TestAccLightsailInstance_basic
=== RUN   TestAccLightsailInstance_name
=== PAUSE TestAccLightsailInstance_name
=== CONT  TestAccLightsailInstance_basic
=== CONT  TestAccLightsailInstance_name
--- PASS: TestAccLightsailInstance_basic (54.17s)
--- PASS: TestAccLightsailInstance_name (151.66s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/lightsail	157.834s
```

**Comment by ewbankkit** [comment]
@danielcotton Thanks for the contribution :tada: :clap:.

**Pull request merged**